### PR TITLE
[Feature] Migrate CQL from D4RLExperienceReplay to MinariExperienceReplay + fix W&B logging and SLURM usage

### LINKED ISSUE #3034: [Feature Request] Migrate D4RLExperienceReplay to MinariExperienceReplay following official deprecation
## Motivation

D4RLExperienceReplay is currently used in `torchrl` to load offline datasets for algorithms like CQL. However, the maintainers of D4RL have officially deprecated this dataset source, stating:

> "All offline datasets in D4RL have been moved to [Minari](https://minari.farama.org/), and all online environments have been transferred to Gymnasium, MiniGrid, and Gymnasium-Robotics."

To stay aligned with this change and maintain long-term compatibility, torchrl should migrate from `D4RLExperienceReplay` to `MinariExperienceReplay`.

## Solution

Update offline RL training scripts to replace `D4RLExperienceReplay` with [`MinariExperienceReplay`](https://pytorch.org/rl/0.6/reference/generated/torchrl.data.datasets.MinariExperienceReplay.html). This ensures the project continues to work with actively maintained offline RL datasets and avoids future breakage.

## Alternatives

- Continue using `D4RLExperienceReplay` temporarily, but this will eventually lead to broken compatibility as D4RL is no longer maintained.
- Implement a transitional wrapper that abstracts over both replay sources (less ideal given clear deprecation).

## Additional context

This request is backed by the official communication in the [Farama Foundation's D4RL GitHub repository](https://github.com/Farama-Foundation/d4rl) and their migration plan.

The initial work has already been implemented for CQL in a related PR. Further migration can follow in subsequent steps for other offline RL agents.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)

### PR DESCRIPTION
**Pull request by Ibinarriaga8**
## Description

This PR updates the offline CQL sota-implementation by migrating from the deprecated D4RLExperienceReplay to the new MinariExperienceReplay, following the recent deprecation notice from the D4RL maintainers. It also includes fixes and improvements for tooling and script robustness.

## Motivation and Context

D4RL’s official repositories have announced that:

> All offline datasets in D4RL have been moved to [Minari](https://minari.farama.org/), and all online environments have been transferred to Gymnasium, MiniGrid, and Gymnasium-Robotics.

To stay aligned with the evolving ecosystem and maintain long-term compatibility, this PR updates the offline dataset interface to use MinariExperienceReplay.

Additionally, it fixes minor bugs affecting W&B logging and SLURM job script behavior, while also ensuring full compliance with the repo’s linting and formatting standards.

No functional changes have been introduced beyond the mentioned components.

Fixes #3034

- [x] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: sota-implementations/cql/utils.py [hunk 1]
```diff
@@ -181,13 +182,13 @@ def make_replay_buffer(
 
 
 def make_offline_replay_buffer(rb_cfg):
-    data = D4RLExperienceReplay(
+    data = MinariExperienceReplay(
```
Comment (Ibinarriaga8) [inline sota-implementations/cql/utils.py:184]: Replaced D4RLExperienceReplay with MinariExperienceReplay following official deprecation of D4RL datasets.
See: https://github.com/Farama-Foundation/d4rl

**Review by vmoens — COMMENTED** [review]
Happy with these changes, do you have a run log to share?

**Comment by Ibinarriaga8** [comment]
> Happy with these changes, do you have a run log to share?

Yes, you can find the run here:
https://wandb.ai/202206789-universidad-pontificia-comillas/torchrl_example_cql/workspace?nw=nwuser202206789

It includes a run of the updated offline CQL sota-implementation using MinariExperienceReplay on mujoco/hopper/expert-v0 dataset.

**Comment by vmoens** [comment]
I cannot see it unfortunately

**Comment by Ibinarriaga8** [comment]
[MinariExperienceReplay Report.pdf](https://github.com/user-attachments/files/21060392/MinariExperienceReplay.Report.pdf)

You can see the logs from W&B in this pdf

**Comment by Ibinarriaga8** [comment]
This is a comparison between the training logs using **Minari and the D4RL Replay Buffer**. Although they generally display similar behavior, D4RL shows higher noise in the Q-function loss during training.

[D4RL v Minari Report.pdf](https://github.com/user-attachments/files/21104487/D4RL.v.Minari.Report.pdf)
[W&B Report](https://wandb.ai/202206789-universidad-pontificia-comillas/torchrl_example_cql/reports/D4RL-vs-Minari-Experience-Replay--VmlldzoxMzQ4NTYxNA)

While the overall training behavior remains consistent, the increased Q-function noise with D4RL suggests Minari may provide slightly more stable learning curves

**Comment by Ibinarriaga8** [comment]
@vmoens I can see sota-checks were succesful: https://github.com/pytorch/rl/actions/runs/16122638416/workflow

However, can you clarify what you mean by sota-runs implementations failed? Is this referring to a different workflow or step beyond the sota-check?

**Review by vmoens — APPROVED** [review]
Ok let's merge then!
